engine/src/main/java/org/terasology/world/block/items/BlockItemSystem.java

```diff
@@ -112,6 +121,28 @@ public void onPlaceBlock(ActivateEvent event, EntityRef item) {
         }
     }
 
+    /**
+     * Returns the position at which the block side was hit, relative to the side.
+     * <p/>
+     * Example: The front side was hit right in the center.
+     * The result will be (0.5, 0.5), representing the relative hit position on the side's surface.
+     * @param hitPosition the hit position
+     * @param blockPosition the block position relative to its center (block (0, 0, 0) has block position (0.5, 0.5, 0.5))
+     * @return the 2D hit position relative to the side that was hit
+     */
+    private Vector2f getSideHitPosition(Vector3f hitPosition, Vector3f blockPosition) {
```

Comment:
This assumes a cubic block, right? I'm totally fine with that, but maybe we should mention that in the doc comment?

Reply:
I added some explanation.